feature/home/src/main/java/com/epmedu/animeal/home/HomeScreenUI.kt

```diff
@@ -19,15 +19,77 @@ import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
 import com.epmedu.animeal.extensions.launchAppSettings
 import com.epmedu.animeal.foundation.button.AnimealButton
 import com.epmedu.animeal.foundation.switch.AnimealSwitch
+import com.epmedu.animeal.foundation.theme.AnimealTheme
 import com.epmedu.animeal.resources.R
 import com.google.accompanist.permissions.ExperimentalPermissionsApi
 import com.google.accompanist.permissions.isGranted
 import com.google.accompanist.permissions.rememberPermissionState
 import com.google.accompanist.permissions.shouldShowRationale
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapView
+import com.mapbox.maps.ResourceOptions
+import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.plugin.scalebar.scalebar
+
+@Composable
+internal fun HomeScreenUI(state: HomeState) {
+    CheckLocationPermission {
+        Box(modifier = Modifier.fillMaxSize()) {
+            MapboxMap(state)
+            AnimealSwitch(
+                modifier = Modifier
+                    .align(alignment = Alignment.TopCenter)
+                    .padding(top = 24.dp),
+                onTabSelected = {}
+            )
+        }
+    }
+}
+
+@Composable
+private fun MapboxMap(state: HomeState) {
+    val mapBoxView = mapView(state.mapBoxPublicKey, state.mapBoxStyleUri)
+
+    val onIndicatorPositionChangedListener = OnIndicatorPositionChangedListener {
+        mapBoxView.getMapboxMap().setCamera(CameraOptions.Builder().center(it).build())
```

Comment:
If you want to set camera on current location, you could set initial currentLocation to null in the state, add flag, remember it and check both like this
var gotInitialLocation by rememberSaveable { mutableStateOf(false) }
LaunchedEffect(Unit) { gotInitialLocation = false }
inside update callback of AndroidView:
if (state.currentLocation != null && gotInitialLocation.not()) {
gotInitialLocation = true
setCameraOnLocation(state.currentLocation)
}

Reply:
Fixed